File: examples/basic-service/src/test/scala/com/oracle/infy/qa/BasicServiceTest.scala
package com.oracle.infy.qa

import akka.testkit.TestProbe
import com.oracle.infy.wookiee.service.messages.GetMetaDetails
import com.oracle.infy.wookiee.service.meta.ServiceMetaDetails
import com.oracle.infy.wookiee.test.BaseWookieeTest
import com.typesafe.config.{Config, ConfigFactory}
import org.scalatest.matchers.should.Matchers
import org.scalatest.wordspec.AnyWordSpecLike

class BasicServiceTest extends BaseWookieeTest with AnyWordSpecLike with Matchers {
  override def config: Config = ConfigFactory.empty()
  override def servicesMap: Option[Map[String, Class[BasicService]]] = Some(Map("base" -> classOf[BasicService]))

  "BasicService" should {
    "start itself up" in {
      val probe = TestProbe()
      val testService = testWookiee.getService("base")
      assert(testService.isDefined, "Basic Service was not registered")

      probe.send(testService.get, GetMetaDetails)
      ServiceMetaDetails(false) shouldEqual probe.expectMsg(ServiceMetaDetails(false))
    }
  }
}

File: examples/java-service/src/test/java/com/oracle/infy/qa/BasicServiceTest.java
package com.oracle.infy.qa;

import com.oracle.infy.wookiee.service.WookieeService;
import com.oracle.infy.wookiee.service.meta.WookieeServiceMeta;
import com.oracle.infy.wookiee.test.BaseWookieeTest;
import com.typesafe.config.Config;
import com.typesafe.config.ConfigFactory;
import org.junit.Test;
import scala.Option;
import scala.collection.immutable.HashMap;
import scala.collection.immutable.Map;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

public class BasicServiceTest implements BaseWookieeTest {
  Map<String, Class<? extends WookieeService>> servicesMap = new HashMap<String, Class<BasicService>>().updated("base", BasicService.class);
  @Override
  public Option<Map<String, Class<? extends WookieeService>>> servicesMap() {
    return Option.apply(servicesMap);
  }

  @Test
  public void testBasicServiceStartsItselfUp() {
    Option<WookieeServiceMeta> testService = testWookiee().getService("base");
    assertNotNull("Basic Service was not registered", testService);
    assertEquals(testService.get().name(), "base");
  }
}
